tests/sort_tests.rs

```diff
@@ -136,7 +132,7 @@ fn complex_sort() {
                         |b,i,e| merge(b, e)
                       )
                     ),
-                    |x:{i32, i32}| x.$0
+                    |x:{i32, i32}, y:{i32, i32}| compare(x.$0, y.$0)
```

Comment:
Maybe swap the zip order so the sort actually does something? I think right now its already sorted by the first struct element, so nothing will change.

Reply:
Probably worth adding a few more tests here -- e.g., vectors of floats to make sure memcmp isn't called at the wrong time, structs with mixed types, structs with structs or vectors, etc. I can write some of these as well

Reply:
Added a few tests.